src/raftstore/store/peer.rs

```diff
@@ -350,6 +355,55 @@ impl Peer {
         Ok(())
     }
 
+    pub fn step(&mut self, m: eraftpb::Message) -> Result<()> {
+        match m.get_msg_type() {
+            eraftpb::MessageType::MsgAppend |
+            eraftpb::MessageType::MsgAppendResponse |
+            eraftpb::MessageType::MsgHeartbeat |
+            eraftpb::MessageType::MsgHeartbeatResponse |
+            eraftpb::MessageType::MsgRequestVote |
```

Comment:
I don't think type checking is necessary here. If `m.get_from() != INVALID_ID`, then peer_heartbeats can be updated.